libs/domains/services/feature/src/lib/service-badge/service-badge.tsx

```diff
@@ -0,0 +1,64 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import { type JobLifecycleTypeEnum } from 'qovery-typescript-axios'
+import { type ComponentPropsWithoutRef, type ElementRef, forwardRef } from 'react'
+import { type IconEnum } from '@qovery/shared/enums'
+import { Icon, Tooltip } from '@qovery/shared/ui'
+import { twMerge, upperCaseFirstLetter } from '@qovery/shared/util-js'
+
+const serviceBadgeVariants = cva(
+  ['relative', 'flex', 'items-center', 'justify-center', 'rounded-full', 'border', 'border-neutral-200'],
+  {
+    variants: {
+      size: {
+        md: ['h-16', 'w-16'],
+        xs: ['h-8', 'w-8'],
+      },
+    },
+  }
+)
+
+const iconVariants = cva('', {
+  variants: {
+    size: {
+      md: ['h-10', 'w-10'],
+      xs: ['h-6', 'w-6'],
+    },
+  },
+})
+
+const typeVariants = cva(['absolute', '-right-0.5', '-top-0.5', 'p-0.5'], {
+  variants: {
+    size: {
+      md: ['h-6', 'w-6'],
+      xs: ['h-4', 'w-4', '-top-1', '-right-1.5'],
+    },
+  },
+})
+
+export interface ServiceBadgeProps
+  extends Omit<ComponentPropsWithoutRef<'div'>, 'size'>,
+    VariantProps<typeof serviceBadgeVariants> {
+  icon: keyof typeof IconEnum
+  type?: keyof typeof JobLifecycleTypeEnum
+}
+
+// TODO: Prepare component to update service icon with custom icon
+export const ServiceBadge = forwardRef<ElementRef<'div'>, ServiceBadgeProps>(function ServiceBadge(
```

Comment:
I think we can improve several points on this component:
1. It's probably the right time to introduce a new UI component `ResourceAvatar` which can represent any resource visual marker. We can have typescript props the guarantee correct typing.
2. I don't like the term `Badge` as it's already something else in the design system.
3. A more composable approach is to have a component `Indicator`, which is only a wrapper to its children and add an indicator at its top right corner. Like the color dot we have but not limited to that case and supports of icons.